trs_filer/ga4gh/trs/server.py

```diff
@@ -2,20 +2,44 @@
 
 from typing import (Optional, Dict, List)
 
-from flask import request
+from flask import (request, current_app)
 from foca.utils.logging import log_traffic
 
 from trs_filer.ga4gh.trs.endpoints.register_tools import (
     RegisterObject,
 )
+from trs_filer.errors.exceptions import NotFound
+import logging
+
+logger = logging.getLogger(__name__)
 
 
 @log_traffic
 def toolsIdGet(
     id: str
 ) -> Dict:
-    """ List one specific tool, acts as an anchor for self references. """
-    return {}  # pragma: no cover
+    """List one specific tool, acts as an anchor for self references.
+    Args:
+        id: A unique identifier of the tool.
+    Returns:
+        Tool object dict corresponding given tool id.
+    Raise:
+        NotFound if no object mapping with given id present.
+    """
+
+    db_collection = (
+        current_app.config['FOCA'].db.dbs['trsStore']
+        .collections['objects'].client
+    )
+
+    obj = db_collection.find_one({"id": id})
+
+    if not obj:
+        logger.info(f"Tool object mapping for id:{id} cannot be found.")
```

Comment:
I think this is redundant, because we are using the decorator that will log both request and response. It will be obvious from the response that a `404` was raised and on what call, because the request is included again in the response, I think. So you can just remove that. And then also remove the `logging` import and `logger` setup, as it looks like you don't need it anywhere else at the moment...